comit/src/network/protocols/orderbook.rs

```diff
@@ -225,136 +415,190 @@ impl NetworkBehaviourEventProcess<GossipsubEvent> for Orderbook {
-    }
+        let delay = Duration::from_secs(2);
 
-    pub fn take_with_identities(
-        &mut self,
-        order_id: OrderId,
-        refund_identity: crate::bitcoin::Address,
-        redeem_identity: identity::Ethereum,
-    ) -> anyhow::Result<Order> {
-        self.identities
-            .insert(order_id, (refund_identity, redeem_identity));
-        let order = match self.orders.remove(&order_id) {
-            Some(order) => order,
-            None => return Err(anyhow::Error::from(OrderbookError::OrderNotFound(order_id))),
+        let (mut alice_swarm, _, _alice_peer_id) = new_swarm(Orderbook::new);
+        let (mut bob_swarm, bob_addr, bob_peer_id) = new_swarm(Orderbook::new);
+        connect(&mut alice_swarm, &mut bob_swarm).await;
+
+        let order = create_order(bob_peer_id.clone(), bob_addr.clone());
+
+        // act
+
+        // Trigger subscription.
+        poll_with_delay(&mut alice_swarm, delay).await;
+        poll_with_delay(&mut bob_swarm, delay).await;
+
+        let _ = bob_swarm
+            .make(order.clone())
+            .expect("order id should exist");
+
+        // Trigger publish and receipt of order.
+        poll_with_delay(&mut bob_swarm, delay).await;
+        poll_with_delay(&mut alice_swarm, delay).await;
+
+        let order = alice_swarm
+            .get_orders()
+            .first()
+            .cloned()
+            .expect("Alice has no orders");
+
+        tracing::info!("alice received order");
+
+        alice_swarm.take(order.id).expect("failed to take order");
+
+        // Trigger request/response messages.
+        poll_with_delay(&mut alice_swarm, delay).await;
+
+        let bob_event = tokio::time::timeout(delay, bob_swarm.next())
+            .await
+            .expect("failed to get TakeOrderRequest event");
+
+        let (peer_id, channel, order_id) = match bob_event {
+            BehaviourOutEvent::TakeOrderRequest {
+                peer_id,
+                response_channel,
+                order_id,
+            } => (peer_id, response_channel, order_id),
+            _ => panic!("unexepected bob event"),
         };
-        Ok(order)
-    }
+        bob_swarm.confirm(peer_id, order_id, channel);
 
-    pub fn get_orders(&self) -> Vec<Order> {
-        #[allow(clippy::map_clone)]
-        self.orders.values().map(|order| order.clone()).collect()
-    }
+        let alice_event = tokio::time::timeout(delay, alice_swarm.next())
+            .await
+            .expect("failed to get TakeOrderRequest event");
 
-    pub fn get_order(&self, order_id: &OrderId) -> Option<Order> {
-        if let Some(order) = self.orders.get(order_id) {
-            Some(order.clone())
-        } else {
-            None
+        match alice_event {
+            // TODO: Assert order id and swap id are correct
+            BehaviourOutEvent::TakeOrderConfirmation { .. } => (),
+            _ => panic!("alice failed to receive take order confirmation"),
         }
     }
```

Comment:
Have you thought of an API where Bob also gets an event after he confirmed the order?

I am surprised this test passes without polling Bob any further :D